credentials/insecure/insecure.go

```diff
@@ -22,11 +22,17 @@ package insecure
 
 import (
 	"context"
+	"encoding/json"
 	"net"
 
 	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/xds/bootstrap"
```

Comment:
This adds an xds dependency to `insecure`.

Should the builders be defined in `xds/internal/bootstrap` to avoid adding the dependencies?
@dfawley WDYT?

Reply:
> This adds an xds dependency to insecure.

In reality, this is just a trivial registry, not something that has any meaningful size behind it.

That said, I agree the name is scary, and the potential for future additions due to the generic name could cause a problem.  I guess I'm fine with putting these implementations elsewhere to avoid that.  Maybe just `xds/bootstrap` (nothing gets exported anyway)?

Reply:
I would prefer `xds/internal/bootstrap` over `xds/bootstrap`, just so we keep all the implementation together.
There's no difference either way.

Reply:
Done, moved it back to xds/internal/bootstrap.